Arm/State.lean

```diff
@@ -1022,3 +1074,32 @@ by_cases h : ix < base
   · simp only [h₂, ↓reduceIte, BitVec.getLsb_extractLsByte]
 
 end Memory
+
+/-! ## Helper lemma for `AxEffects` -/
+
+theorem Memory.eq_of_read_mem_bytes_eq {m₁ m₂ : Memory}
```

Comment:
Ah, this reminds me: we have a file [`StateEq.lean`](https://github.com/leanprover/LNSym/blob/main/Arm/StateEq.lean) which has such eq lemmas. Unfortunately, I didn't imported it anywhere, so it didn't keep up with changes to LNSym.

I'll create an issue to update that file.

Reply:
I didn't know about that file. Let's keep these lemmas here for now, and mention in that issue that we should get rid of the hlper lemmas in `State.lean` once `StateEq.lean` builds again